File: 프로그래머스/카펫.py
def solution(brown, yellow):
    answer = []
    n=(brown+4)//2
    print(n)
    for i in range(1, n):
        x=i
        y=n-i
        if (x-2)*(y-2)==yellow:
            width=max(x,y)
            height=min(x,y)
            answer.append(width)
            answer.append(height)
            break
    
    return answer

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int tmp=brown/2+2;
        for (int i=1; i<tmp/2+1; i++){
            if ((i-2)*(tmp-i-2)==yellow){
                answer[0]=Math.max(i, tmp-i);
                answer[1]=Math.min(i, tmp-i);
                break;
            }
        }
        return answer;
        
        //방정식
        //4-brown=-2(x+y) 구해놓고
        //1,2,3, 증가하면 yellow값 맞는지
    }
}
